src/modelmanager.cpp

```diff
@@ -444,13 +444,19 @@ Status ModelManager::loadModelsConfig(rapidjson::Document& configJson, std::vect
     for (const auto& configs : itr->value.GetArray()) {
         ModelConfig modelConfig;
         auto status = modelConfig.parseNode(configs["config"]);
+        const auto modelName = modelConfig.getName();
```

Comment:
Is it safe? You're reading name from the config before checking the status. What if parsing fails?

Reply:
Can you add a commend for this line that this call is safe because if model name is not available it would fail a lot earlier ? Just to avoid confusion in the future.